benchmarks/common.py

```diff
@@ -792,6 +792,71 @@ def decay_batch_exp(self, batch_size, factor=0.5, divisor=2):
             out_batch_size = batch_size - 1
         return max(0, int(out_batch_size))
 
+    def measure_latency_and_memory(self, device, model_name, model_iter_fn, backends):
+        get_peak_memory = lambda: torch.cuda.max_memory_allocated() / 10**9
+        backends.insert(0, "baseline")
+
+        latencies = list()
+        mems = list()
+
+        def mem_experiment(backend):
+            try:
+                # Get the context
+                if backend == "baseline":
+                    ctx = NullContext()
+                else:
+                    ctx = torchdynamo.optimize(backend)
+
+                # Reset and warmup
+                torchdynamo.reset()
+                torch.cuda.empty_cache()
+                t0 = time.perf_counter()
+                with ctx:
+                    model_iter_fn(model, example_inputs)
+                    model_iter_fn(model, example_inputs)
+                    model_iter_fn(model, example_inputs)
+                t1 = time.perf_counter()
+                latency = t1 - t0
+
+                # Measure memory
+                torch.cuda.reset_peak_memory_stats()
+                with ctx:
+                    model_iter_fn(model, example_inputs)
+                peak_memory = get_peak_memory()
+            except:
+                latency = 0
+                peak_memory = 0
+
+            latencies.append(latency)
+            mems.append(peak_memory)
+
+        assert device == "cuda"
+        try:
+            device, name, model, example_inputs, batch_size = self.load_model(
+                device,
+                model_name,
+                self._args.training,
+                self._args.use_eval_mode,
+            )
+        except NotImplementedError:
+            logging.warn(f"{model_name} failed to load")
+
+        # Collect the compile and memory latency for backend
+        for backend in backends:
+            mem_experiment(backend)
```

Comment:
I don't know CUDA memory well, but retreating back to CPU memory / RAM, I would consider giving each backend its own process. What would happen here / how would our numbers look if a backend leaked?

Reply:
Yeah, thats good point. Its relevant to GPU memory as well. Will run each backend in isolation.